### PR TITLE
meta: fix meta_inrange_{get,add,del}

### PR DESCRIPTION
**Pull request by MaskRay**
Note meta_inrange_get is only called by r_meta_find_in, which is only used when asm.meta=false (while the default is true). This is very uncommon and thus we don't see issues.

### CODE REVIEW

**Comment by MaskRay** [comment]
BTW, `asm.meta=false` (its description is also dubious) has very weird disasm output.

File: libr/anal/meta.c [hunk 1]
```diff
@@ -27,16 +27,18 @@
...
-#define META_RANGE_SIZE 0xfff
 #undef DB
 #define DB a->sdb_meta
 
 static char *meta_inrange_get (RAnal *a, ut64 addr, int size) {
+	if (size <= 0) {
+		return NULL;
+	}
 	ut64 base = META_RANGE_BASE (addr);
-	ut64 base2 = META_RANGE_BASE (addr + size) + 1;
```
Comment (ret2libc) [inline libr/anal/meta.c]: Why +1 becomes -1? Also, why the <= instead of <? I think in a for loop it’s usually more common to use <

Comment (MaskRay) [reply libr/anal/meta.c]: This is different. 

It is common to do: `for (int i = lo; i < hi; i++)`
But if you want to mark high bits of every number in [lo, hi)

``for (int i = lo; i < hi; i++) unique i>>8`

is equivalent to `for (int i = lo>>8; i <= hi-1 >> 8; i++)`

**Comment by radare** [comment]
its green so lets just merge :3 wonder if you have any test to reproduce the problem you are fixing here

**Comment by MaskRay** [comment]
@radare I have asked what `asm.meta=false` does. The search code is only relevant in that case.

**Comment by radare** [comment]
what it does is to put the .hex , .string, etc..inside the opstring instead of messing all the logic inside the disasm to extend the metadata in the disassembly loop.

its a simplification that works pretty well but nobody but me have tested it. so it needs more testing and probably bugfixing to get it working fine

File: libr/anal/meta.c [hunk 2]
```diff
@@ -27,16 +27,18 @@
...
+	ut64 base2 = META_RANGE_BASE (addr + size - 1);
 	char *res = NULL;
 	// return string array of all the offsets where there are stuff
-	for (; base < base2; base += META_RANGE_SIZE) {
+	for (; base <= base2; base++) {
 		const char *key = sdb_fmt ("range.0x%"PFMT64x, base);
```
Comment (radare) [inline libr/anal/meta.c]: uhmf so the base now is no longer sized in RANGE_SIZE, but 1 by 1?

File: libr/anal/meta.c [hunk 3]
```diff
@@ -52,13 +54,15 @@ static char *meta_inrange_get (RAnal *a, ut64 addr, int size) {
 }
 
 static bool meta_inrange_add (RAnal *a, ut64 addr, int size) {
+	if (size <= 0) {
```
Comment (radare) [inline libr/anal/meta.c]: just < 1

File: libr/anal/meta.c (hunk 2 above)
Comment (MaskRay) [reply libr/anal/meta.c]: If I understand it correctly, range.0xI is set if there is metadata in [i*4096, (i+1)*4096).

The updated code achives this, before, it was incorrect.

File: libr/anal/meta.c (hunk 3 above)
Comment (MaskRay) [reply libr/anal/meta.c]: 0 is more natural than 1. `<= 0` is also more akin to what optimized codegen does.

**Pull request merged**